### ISSUE TITLE
Palette entry reports lack index element

### ISSUE THREAD

**Issue by dankamongmen** [contour-terminal/contour#576]
### Contour Terminal version

0.3.0-unreleased-master-17897620

### Installer source

Github: source code cloned

### Operating System

Debian Linux Unstable

### Architecture

x86-64

### Other Software

notcurses 3.0.3/master

### Steps to reproduce

Send a paltte index request to contour. Here I reuqest palette index 0:

` echo -e "\x1b\x5d\x34\x3b\x30\x3b\x3f\x1b\x5c"`

### Expected Behavior

wezterm: `4;0;rgb:0000/0000/0000`
xterm: `^[]4;0;rgb:0000/0000/0000^[\`
kitty: `4;0;rgb:0000/0000/0000`

### Actual Behavior

Contour replies without the palette index in its response:

`4;rgb:1d/1f/21`

This is unlike every other terminal I've tested with, and furthermore makes it difficult to send pipelined palette queries (if you happened to reorder the responses, I'd have a bad time).

### Additional notes

your %02x is no problem -- i handle that -- but i need the palette index in the reply. likewise, end with ST or don't; i don't particularly care.

i will handle these responses so they don't bleed through as user input (see https://github.com/dankamongmen/notcurses/issues/2539), but i can't actually use them without a palette index. it is desirable that i can learn the palette, for a number of reasons.

**Comment by christianparpart** [comment]
@dankamongmen Thanks for messing around with me at 4:30 AM :-D

`s/\x1b\x5d\x34\x3b\x30\x3b\x3f\x1b\x5c/ESC OSC 4 ; 0 ; ? ESC ST` :tada: :partying_face: :balloon:

**Comment by christianparpart** [comment]
@dankamongmen I really want to please you and fix this BUT why the hex 4 xdigits? I mean, I did look at hte source code of xterm and also noticed it's just doing %04x.... but looking at the output, it's abab/cdcd/efef ALL the time this repetitive pattern.
In xterm the red/green/blue struct members are 16-bit wide, so it makes sense, but 16? Maybe I'm tired, but I can't figure why 16 and why this repetitive pattern. 

I've fixed the index problem already (see PR) and would like to merge as soon as I do the %04-solution.

Except that I don't want to do this:

```cpp
reply_with("\033]4;%d;rgb:%02d%02d/%02d%02d/%02d%02d",
           colorIndex,
           red, red,
           green, green,
           blue, blue);
```
this looks like cheating to me but it would fit the pattern observed.

**Comment by j4james** [comment]
The reason for the repetitive patten is because there's a scaling factor of 0x101 when converting from 8-bit components to 16-bits (i.e. 0xAB * 0x101 = 0xABAB). And the reason why the scaling factor is 0x101, is because that guarantees the maximum 8-bit value (0xFF) will scale up to the maximum 16-bit value (0xFFFF). At least that was my understanding - I could be wrong.

**Comment by christianparpart** [comment]
> The reason for the repetitive patten is because there's a scaling factor of 0x101 when converting from 8-bit components to 16-bits (i.e. 0xAB * 0x101 = 0xABAB). And the reason why the scaling factor is 0x101, is because that guarantees the maximum 8-bit value (0xFF) will scale up to the maximum 16-bit value (0xFFFF). At least that was my understanding - I could be wrong.

Good morning! Well, xterm source code isn't really a reference here because it's depending on what XParseColor understands, which is a pitty that a VT sequence depends on other subsystems (Xlib) lol. However, I found out that `%02x` for the 8-bit color components is just fine (as per man page of `XParseColor`), so yeah. I've been change that just now to 16-bit equivalents though too. Thanks :)

**Comment by christianparpart** [comment]
> ### Expected Behavior
> 
> wezterm: `4;0;rgb:0000/0000/0000`
> xterm: `^[]4;0;rgb:0000/0000/0000^[\`
> kitty: `4;0;rgb:0000/0000/0000`

@dankamongmen Why do these three vary in your quote? By accident? I checked locally and they all act like xterm (and now Copntour does to. yay. what a productive weekend) :)

**Comment by christianparpart** [comment]
Closed by #580.
